### PR TITLE
Filtering of service accounts by expired tokens

### PR DESCRIPTION
**Pull request by eleijonmarck**
**What this PR does / why we need it**:
- filtering of service accounts based on expired tokens

fix: not within the scope of the PR
- introducing no service accounts display (to add new service accounts to empty table)

**Which issue(s) this PR fixes**:
https://github.com/grafana/grafana-enterprise/issues/2801

Fixes #

**Special notes for your reviewer**:

https://user-images.githubusercontent.com/7727602/158845698-76aa29b7-4a90-40cb-944b-577c70fa245c.mov

### CODE REVIEW

File: pkg/services/serviceaccounts/api/api.go [hunk 1]
```diff
@@ -218,7 +218,14 @@ func (api *ServiceAccountsAPI) SearchOrgServiceAccountsWithPaging(c *models.ReqC
 	if page < 1 {
 		page = 1
 	}
-	serviceAccountSearch, err := api.store.SearchOrgServiceAccounts(ctx, c.OrgId, c.Query("query"), page, perPage, c.SignedInUser)
+	expiredTokensFilterQuery := c.Query("expiredTokens")
+	// its okay that it fails, it is only filtering that might be weird, but to safe quard against any weird incoming query param
+	onlyWithExpiredTokens, _ := strconv.ParseBool(expiredTokensFilterQuery)
```
Comment (gamab) [inline pkg/services/serviceaccounts/api/api.go:223]:
```suggestion
	// its okay that it fails, it is only filtering that might be weird, but to safe quard against any weird incoming query param
	onlyWithExpiredTokens := c.QueryBool("expiredTokens")
```

File: pkg/services/serviceaccounts/models.go [hunk 2]
```diff
@@ -60,3 +60,27 @@ type ServiceAccountProfileDTO struct {
...
+
+	User *models.SignedInUser
+}
+
+type SearchOrgServiceAccountsQueryResult struct {
+	TotalCount      int64                `json:"totalCount"`
+	ServiceAccounts []*ServiceAccountDTO `json:"orgServiceAccounts"`
+	Page            int                  `json:"page"`
+	PerPage         int                  `json:"perPage"`
+}
```
Comment (gamab) [inline pkg/services/serviceaccounts/models.go:86]: It doesn't seem used, is it?
```suggestion
```

**Review by gamab — APPROVED** [review]
Code looks good to me!
Seems like we introduced unused struct?

File: pkg/services/serviceaccounts/api/api.go (hunk 1 above)
Comment (eleijonmarck) [reply pkg/services/serviceaccounts/api/api.go:223]: I was trying to cover for parsing that failed to create a boolean, but I guess we should expect correct parameters from the frontend query

File: pkg/services/serviceaccounts/models.go (hunk 2 above)
Comment (eleijonmarck) [reply pkg/services/serviceaccounts/models.go:86]: thissss is true! well done to find this. This was part of a bigger refactoring that you too did and remove the query.
I missed to remove them! nice work.

Curious did your ide or just your big brain see this ? 🧠

File: pkg/services/serviceaccounts/api/api.go (hunk 1 above)
Comment (eleijonmarck) [reply pkg/services/serviceaccounts/api/api.go:223]: what happens to QueryBool, if it fails to see a boolean or anything in particular? 🤔

Comment (eleijonmarck) [reply pkg/services/serviceaccounts/api/api.go:223]: it defaults to false I hope >D

Comment (gamab) [reply pkg/services/serviceaccounts/api/api.go:223]: I'm not sure I follow, it seems to me `QueryBool` does internally exactly the same as you are doing:
```go
// QueryBool returns query result in bool type.
func (ctx *Context) QueryBool(name string) bool {
	v, _ := strconv.ParseBool(ctx.Query(name))
	return v
}
```

Comment (eleijonmarck) [reply pkg/services/serviceaccounts/api/api.go]:
```
// ParseBool returns the boolean value represented by the string.
// It accepts 1, t, T, TRUE, true, True, 0, f, F, FALSE, false, False.
// Any other value returns an error.
func ParseBool(str string) (bool, error) {
	switch str {
	case "1", "t", "T", "true", "TRUE", "True":
		return true, nil
	case "0", "f", "F", "false", "FALSE", "False":
		return false, nil
	}
	return false, syntaxError("ParseBool", str)
}
```
thats great

**Pull request merged**